src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1801 @@
+        }
+        else
+        {
+            HandleCommand<Commands::SnapshotStreamDeallocate::DecodableType>(
+                handlerContext,
+                [this](HandlerContext & ctx, const auto & commandData) { HandleSnapshotStreamDeallocate(ctx, commandData); });
+        }
+        return;
+
+    case Commands::SetStreamPriorities::Id:
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Set Stream Priorities");
+        HandleCommand<Commands::SetStreamPriorities::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleSetStreamPriorities(ctx, commandData); });
+        return;
+
+    case Commands::CaptureSnapshot::Id:
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Capture Snapshot image");
+
+        if (!HasFeature(Feature::kSnapshot))
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        }
+        else
+        {
+            HandleCommand<Commands::CaptureSnapshot::DecodableType>(
+                handlerContext,
+                [this](HandlerContext & ctx, const auto & commandData) { HandleCaptureSnapshot(ctx, commandData); });
+        }
+        return;
+    }
+}
+
+void CameraAVStreamMgmtServer::HandleVideoStreamAllocate(HandlerContext & ctx,
+                                                         const Commands::VideoStreamAllocate::DecodableType & commandData)
+{
+    Status status = Status::Success;
+
+    Commands::VideoStreamAllocateResponse::Type response;
+    auto & streamUsage      = commandData.streamUsage;
+    auto & videoCodec       = commandData.videoCodec;
+    auto & minFrameRate     = commandData.minFrameRate;
+    auto & maxFrameRate     = commandData.maxFrameRate;
+    auto & minResolution    = commandData.minResolution;
+    auto & maxResolution    = commandData.maxResolution;
+    auto & minBitRate       = commandData.minBitRate;
+    auto & maxBitRate       = commandData.maxBitRate;
+    auto & minFragmentLen   = commandData.minFragmentLen;
+    auto & maxFragmentLen   = commandData.maxFragmentLen;
+    bool isWaterMarkEnabled = false;
+    bool isOSDEnabled       = false;
+
+    if (HasFeature(Feature::kWatermark))
+    {
+        isWaterMarkEnabled = commandData.watermarkEnabled.ValueOr(false);
+    }
+    else
+    {
+        VerifyOrExit(!commandData.watermarkEnabled.HasValue(), status = Status::InvalidCommand;
+                     ChipLogError(Zcl, "CameraAVStreamMgmt: Failed to allocate Video stream. Unsupported watermark feature"));
+    }
+
+    if (HasFeature(Feature::kOnScreenDisplay))
+    {
+        isOSDEnabled = commandData.OSDEnabled.ValueOr(false);
+    }
+    else
+    {
+        VerifyOrExit(!commandData.OSDEnabled.HasValue(), status = Status::InvalidCommand;
+                     ChipLogError(Zcl, "CameraAVStreamMgmt: Failed to allocate Video stream. Unsupported OnScreenDisplay feature"));
+    }
+
+    // Call the delegate
```

Comment:
What happened to verifying that the passed-in enums have valid valies?  That the min/max constraints of all sorts are satisfied?  All of that needs to happen here, because relying on the delegate to do that checking means it won't happen.

Reply:
Added sanity checks for args for all the commands.